pkg/filenotify/poller.go

```diff
@@ -52,6 +52,8 @@ func (w *filePoller) Add(name string) error {
 	if err != nil {
 		return err
 	}
+	defer f.Close()
```

Comment:
Agreed, this not correct. This file needs to remain open when this function returns.